### PR TITLE
Improve resolution selection and orientation handling for Native Camera on iOS

### PR DESCRIPTION
**Pull request by Alex-MSFT**
This PR improves resolution selection and orientation handling for Native Camera sessions on iOS.  As outlined in #1107 currently we are just taking the highest resolution setting on the camera ignoring the constraints set for maxWidth and maxHeight.   Additionally currently we are not making any adjustments to the outputted video texture for device orientation.  This PR fixes this so the output texture is oriented in a way that aligns with the screen.

Landscape before:
Before landscape

Landscape after:
After landscape

Portrait before:
Portrait Before

Portrait after:
Portrait after

### CODE REVIEW

**Review by CoPrez — APPROVED** [review]

File: Plugins/NativeCamera/Source/Apple/NativeCameraImpl.mm [hunk 1]
```diff
@@ -58,69 +59,138 @@ - (id)init:(std::shared_ptr<Babylon::Plugins::Camera::Impl::ImplData>)implData;
...
-            UNUSED(frontCamera);
-            AVCaptureDevice* captureDevice = [AVCaptureDevice defaultDeviceWithMediaType:AVMediaTypeVideo];
-#endif
-            // Set video capture input: If there a problem initialising the camera, it will give am error.
-            AVCaptureDeviceInput *input = [AVCaptureDeviceInput deviceInputWithDevice:captureDevice error:&error];
+            
+            // If no matching device, throw an error with the message "ConstraintError" which matches the behavior in the browser.
+            if (bestDevice == NULL)
+            {
+                taskCompletionSource.complete(arcana::make_unexpected(std::make_exception_ptr(std::runtime_error{"ConstraintError"})));
```
Comment (ryantrem) [inline Plugins/NativeCamera/Source/Apple/NativeCameraImpl.mm:161]: Maybe spell this out more? Instead of "ConstraintError" something like "Could not find a device configuration matching the requested constraings." Or just copy what WebAPI does?

**Review by ryantrem — APPROVED** [review]

File: Plugins/NativeCamera/Source/Apple/NativeCameraImpl.mm (hunk 1 above)
Comment (Alex-MSFT) [reply Plugins/NativeCamera/Source/Apple/NativeCameraImpl.mm:161]: This is what the WebAPI does :'(, but I can give it a better error message..

File: Plugins/NativeCamera/Source/Apple/NativeCameraImpl.mm [hunk 2]
```diff
@@ -58,69 +59,138 @@ - (id)init:(std::shared_ptr<Babylon::Plugins::Camera::Impl::ImplData>)implData;
...
+            
+            AVCaptureDeviceDiscoverySession* discoverySession = [AVCaptureDeviceDiscoverySession
+                discoverySessionWithDeviceTypes:deviceTypes
+                mediaType:AVMediaTypeVideo position:frontCamera ? AVCaptureDevicePositionFront: AVCaptureDevicePositionBack];
+            for (AVCaptureDevice* device in discoverySession.devices)
+            {
+                for (AVCaptureDeviceFormat* format in device.formats)
+                {
+                    CMVideoFormatDescriptionRef videoFormatRef = static_cast<CMVideoFormatDescriptionRef>(format.formatDescription);
+                    CMVideoDimensions dimensions = CMVideoFormatDescriptionGetDimensions(videoFormatRef);
```
[the remark this answers is not in the document]
Comment (Alex-MSFT) [reply Plugins/NativeCamera/Source/Apple/NativeCameraImpl.mm:125]: Will do a pass to make this uniform for both my code and the existing code.

File: Plugins/NativeCamera/Source/Apple/NativeCameraImpl.mm (hunk 1 above)
Comment (ryantrem) [reply Plugins/NativeCamera/Source/Apple/NativeCameraImpl.mm:161]: Oh well, if it matches WebAPI then I guess that's something!

File: Plugins/NativeCamera/Source/Apple/NativeCameraImpl.mm [hunk 3]
```diff
@@ -58,69 +61,146 @@ - (id)init:(std::shared_ptr<Babylon::Plugins::Camera::Impl::ImplData>)implData;
     {
     }
 
-    arcana::task<void, std::exception_ptr> Camera::Impl::Open(uint32_t /*width*/, uint32_t /*height*/, bool frontCamera)
+    arcana::task<void, std::exception_ptr> Camera::Impl::Open(uint32_t maxWidth, uint32_t maxHeight, bool frontCamera)
     {
+        if (maxWidth == 0 || maxWidth > INT32_MAX) {
+            maxWidth = INT32_MAX;
```
Comment (ryantrem) [inline Plugins/NativeCamera/Source/Apple/NativeCameraImpl.mm:67]: std::numeric_limits<uint32_t>::max()

File: Plugins/NativeCamera/Source/Apple/NativeCameraImpl.mm [hunk 4]
```diff
@@ -58,69 +61,147 @@ - (id)init:(std::shared_ptr<Babylon::Plugins::Camera::Impl::ImplData>)implData;
     {
     }
 
-    arcana::task<void, std::exception_ptr> Camera::Impl::Open(uint32_t /*width*/, uint32_t /*height*/, bool frontCamera)
+    arcana::task<void, std::exception_ptr> Camera::Impl::Open(uint32_t maxWidth, uint32_t maxHeight, bool frontCamera)
     {
+        if (maxWidth == 0 || maxWidth > std::numeric_limits<int32_t>::max()) {
+            maxWidth = std::numeric_limits<int32_t>::max();
```
Comment (ryantrem) [inline Plugins/NativeCamera/Source/Apple/NativeCameraImpl.mm:67]: Why int32_t instead of uint32_t?

Comment (Alex-MSFT) [reply Plugins/NativeCamera/Source/Apple/NativeCameraImpl.mm:67]: It's an unreasonable resolution either way.  I'm giving overhead to prevent overflow because we might add both maxWidth and maxHeight on line 135 below:
                    uint32_t dimDiff{(maxWidth - dimensions.width) + (maxHeight - dimensions.height)};

If Apple ever releases a camera with 4294967295 x 4294967295 resolution we might be in trouble!

File: Apps/Playground/Scripts/experience.js [hunk 5]
```diff
@@ -63,8 +63,11 @@ CreateBoxAsync(scene).then(function () {
 //BABYLON.SceneLoader.AppendAsync("https://raw.githubusercontent.com/KhronosGroup/glTF-Sample-Models/master/2.0/ClearCoatTest/glTF/ClearCoatTest.gltf").then(function () {
     BABYLON.Tools.Log("Loaded");
 
-    scene.createDefaultCamera(true, true, true);
-    scene.activeCamera.alpha += Math.PI;
+    // This creates and positions a free camera (non-mesh)
+    var camera = new BABYLON.FreeCamera("camera1", new BABYLON.Vector3(0, 1, -10), scene);
+
+    // This targets the camera to scene origin
+    camera.setTarget(new BABYLON.Vector3(0, 1, 0));
```
Comment (bghgary) [inline Apps/Playground/Scripts/experience.js:70]: What's the reason for this change? Input is no longer working because of this 😕

Comment (CoPrez) [reply Apps/Playground/Scripts/experience.js:70]: This issue has resolved with #1127
